integration/test_auth.py

```diff
@@ -55,6 +59,35 @@ def _random_id():
 def _random_phone():
     return '+1' + ''.join([str(random.randint(0, 9)) for _ in range(0, 10)])
 
+def _reset_password(oob_code, new_password, api_key):
+    body = {'oobCode': oob_code, 'newPassword': new_password}
+    params = {'key' : api_key}
+    resp = requests.request('post', _password_reset_url, params=params, json=body)
+    resp.raise_for_status()
+    return resp.json().get('email')
+
+def _verify_email(oob_code, api_key):
+    body = {'oobCode': oob_code}
+    params = {'key' : api_key}
+    resp = requests.request('post', _verify_email_url, params=params, json=body)
+    resp.raise_for_status()
+    return resp.json().get('email')
+
+def _sign_in_with_email_link(email, oob_code, api_key):
+    body = {'oobCode': oob_code, 'email': email}
+    params = {'key' : api_key}
+    resp = requests.request('post', _email_sign_in_url, params=params, json=body)
+    resp.raise_for_status()
+    return resp.json().get('idToken')
+
+def _validate_link_url(link, check_continue_url=True):
```

Comment:
Lets remove the assertion from this method, and return the entire `query_dict` from it. Then lets rename the method to `extract_link_params()` or something similar.

Callers can decide how to deal with the returned dict.

Reply:
Done.